umap/static/umap/js/modules/urls.mjs

```diff
@@ -0,0 +1,38 @@
+export default class URLs {
+    constructor(serverUrls) {
+        this.urls = serverUrls;
+    }
+
+    get(urlName, params) {
+        if (typeof params !== 'object')
+            params = { map_id: params };
+
+        if (typeof this[urlName] === "function")
+            return this[urlName](params);
+
+        if (this.urls.hasOwnProperty(urlName)) {
+            console.log("urls", this.urls, urlName, params);
+            return L.Util.template(this.urls[urlName], params);
+        } else {
+            throw `Unable to find a URL for route ${urlName}`;
+        }
+    }
+
+    // Update if map_id is passed, create otherwise.
+    map_save(params) {
+        if (typeof params !== 'object')
+            params = { map_id: params };
+        if (params.map_id)
+            return this.get("map_update", params.map_id);
+        return this.get("map_create");
+    }
+
+    // Update the layer if pk is passed, create otherwise.
+    datalayer_save(params) {
+        if (typeof params !== 'object')
```

Comment:
Why not always passing params as an object ?

Reply:
It allows to make the calls easier.

```urls.get('route_name', map_id)``` rather than ```urls.get('route_name', {map_id: map_id})```

Reply:
This piece of check seems to be here and there.
Would it make sense to have a uniformizing function such as

```
function getMapObject(idOrObject) {
  if (!idOrObject.map_id)
    return { map_id: idOrObject }
  else
    return idOrObject
}
```

Then calling it whenever we need to check about the `params` object|string

Reply:
I guess that's because I'm trying to mimic how python works. I've made changes to always pass an object.